mobile_config/tests/integrations/gateway_db.rs

```diff
@@ -0,0 +1,309 @@
+    g2.hash = "y".into();
+    g2.insert(&pool).await?;
+
+    // min_updated_at = t2 => expect empty
+    let s = Gateway::stream_by_addresses(&pool, vec![a1.clone(), a2.clone()], t2);
+    pin_mut!(s);
+    assert!(s.next().await.is_none());
+
+    // bump g1.updated_at to t2
+    let mut g1b = g1.clone();
+    g1b.updated_at = t2;
+    g1b.refreshed_at = t0;
+    g1b.insert(&pool).await?;
+
+    // now we should see g1 only
+    let s = Gateway::stream_by_addresses(&pool, vec![a1.clone(), a2.clone()], t2);
+    pin_mut!(s);
+    let first = s.next().await.expect("one row");
+    assert_eq!(first.address.as_ref(), a1.as_ref());
+    assert!(s.next().await.is_none());
+
+    Ok(())
+}
+
+#[sqlx::test]
+async fn stream_by_types_filters_by_min_date(pool: PgPool) -> anyhow::Result<()> {
+    let t0 = Utc.with_ymd_and_hms(2025, 1, 1, 0, 0, 0).unwrap();
+    let t1 = Utc.with_ymd_and_hms(2025, 1, 2, 0, 0, 0).unwrap();
+    let t2 = Utc.with_ymd_and_hms(2025, 1, 3, 0, 0, 0).unwrap();
+
+    // Insert: two WifiIndoor (t0, t2) and one WifiOutdoor (t1)
+    gw(pk_binary(), GatewayType::WifiIndoor, t0)
+        .insert(&pool)
+        .await?;
+    gw(pk_binary(), GatewayType::WifiOutdoor, t1)
+        .insert(&pool)
+        .await?;
+    gw(pk_binary(), GatewayType::WifiIndoor, t2)
+        .insert(&pool)
+        .await?;
+
+    // min_date = t2, types = WifiIndoor; expect only the t2 indoor
+    let s = Gateway::stream_by_types(&pool, vec![GatewayType::WifiIndoor], t2, None);
+    pin_mut!(s);
+    let first = s.next().await.expect("row expected");
+    assert_eq!(first.gateway_type, GatewayType::WifiIndoor);
+    assert_eq!(first.created_at, t2);
+    assert!(s.next().await.is_none());
+
+    Ok(())
+}
+
+#[sqlx::test]
+async fn stream_by_types_optional_location_changed_filter(pool: PgPool) -> anyhow::Result<()> {
+    let t0 = Utc.with_ymd_and_hms(2025, 2, 1, 0, 0, 0).unwrap();
+    let t1 = Utc.with_ymd_and_hms(2025, 2, 3, 0, 0, 0).unwrap();
+
+    // Indoor @ t0 and Indoor @ t1; set location_changed_at accordingly
+    let a_addr = pk_binary();
+    let mut a = gw(a_addr.clone(), GatewayType::WifiIndoor, t0);
+    a.location_changed_at = Some(t0);
+    a.insert(&pool).await?;
+
+    let b_addr = pk_binary();
+    let mut b = gw(b_addr.clone(), GatewayType::WifiIndoor, t1);
+    b.location_changed_at = Some(t1);
+    b.insert(&pool).await?;
+
+    // min_date = t0 (so date doesn't exclude); set min_location_changed_at = t1
+    let s = Gateway::stream_by_types(&pool, vec![GatewayType::WifiIndoor], t0, Some(t1));
+    pin_mut!(s);
+
+    // Expect only the one with location_changed_at >= t1 (i.e., `b`)
+    let first = s.next().await.expect("row expected");
+    assert_eq!(first.address.as_ref(), b_addr.as_ref());
+    assert!(s.next().await.is_none());
+
+    Ok(())
+}
+
+fn now_truncated() -> chrono::DateTime<Utc> {
+    let n = Utc::now();
+    let micros = n.timestamp_subsec_micros();
+    n.with_nanosecond(micros * 1_000).unwrap()
+}
```

Comment:
https://github.com/helium/oracles/blob/main/reward_index/tests/integrations/common/mod.rs#L18-L23

We're doing something like this in a few places.
Might be a good time to consolidate, or update our CI postgres to be consistent with prod.